### PR TITLE
Throw DeserializationException on invalid json/enum case/date format. Add tests.

### PR DESCRIPTION
**Pull request by sakulb**

### CODE REVIEW

File: src/Handler/Handlers/EnumHandler.php [hunk 1]
```diff
@@ -42,13 +43,23 @@ public function deserialize(mixed $value, Metadata $metadata): UnitEnum
             if ($enumValue instanceof BackedEnum) {
                 return $enumValue;
             }
+            throw new DeserializationException(sprintf(
+                'Cannot deserialize value "%s" into a BackedEnum. Possible options are: "%s".',
+                $value,
+                implode(', ', array_column($metadata->type::cases(), 'value'))
```
Comment (pulzarraider) [inline src/Handler/Handlers/EnumHandler.php:49]:
```suggestion
                implode('", "', array_column($metadata->type::cases(), 'value'))
```

File: src/Handler/Handlers/EnumHandler.php [hunk 2]
```diff
@@ -42,13 +43,23 @@ public function deserialize(mixed $value, Metadata $metadata): UnitEnum
...
         if (is_a($metadata->type, UnitEnum::class, true)) {
             foreach ($metadata->type::cases() as $case) {
                 if ($value === $case->name) {
                     return $case;
                 }
             }
+            throw new DeserializationException(sprintf(
+                'Cannot deserialize value "%s" into a UnitEnum. Possible options are: "%s".',
+                $value,
+                implode(', ', array_column($metadata->type::cases(), 'name'))
```
Comment (pulzarraider) [inline src/Handler/Handlers/EnumHandler.php:61]:
```suggestion
                implode('", "', array_column($metadata->type::cases(), 'name'))
```

File: src/Service/JsonDeserializer.php [hunk 3]
```diff
@@ -5,9 +5,11 @@
 namespace AnzuSystems\SerializerBundle\Service;
 
 use AnzuSystems\SerializerBundle\Exception\SerializerException;
+use AnzuSystems\SerializerBundle\Exception\DeserializationException;
 use AnzuSystems\SerializerBundle\Handler\HandlerResolver;
 use AnzuSystems\SerializerBundle\Metadata\MetadataRegistry;
 use Doctrine\Common\Collections\Collection;
+use \JsonException;
```
Comment (pulzarraider) [inline src/Service/JsonDeserializer.php:12]:
```suggestion
use JsonException;
```

File: tests/AbstractTestCase.php [hunk 4]
```diff
@@ -0,0 +1,27 @@
+<?php
+declare(strict_types=1);
```
Comment (pulzarraider) [inline tests/AbstractTestCase.php:2]: Please add here one more line.

File: tests/AnzuTestKernel.php [hunk 5]
```diff
@@ -0,0 +1,35 @@
+<?php
+declare(strict_types=1);
```
Comment (pulzarraider) [inline tests/AnzuTestKernel.php:2]: Please add here one more line.

File: tests/bootstrap.php [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+$kernel = new AnzuTestKernel('test', false);
+$kernel->boot();
+
+$app = new Application($kernel);
+$app->setAutoExit(false);
+
+$output = new ConsoleOutput();
+
+# Clear cache
```
Comment (pulzarraider) [inline tests/bootstrap.php:23]:
```suggestion
// Clear cache
```

File: tests/bootstrap.php [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+$input = new ArrayInput([
+    'command' => 'cache:clear',
+    '--no-warmup' => true,
+    '--env' => getenv('APP_ENV'),
+]);
+//$input->setInteractive(false);
+$app->run($input, $output);
+
+# Database drop
```
Comment (pulzarraider) [inline tests/bootstrap.php:31]:
```suggestion
// Database drop
```

**Review by pulzarraider — APPROVED** [review]
Good job.

File: tests/AbstractTestCase.php (hunk 4 above)
Comment (sakulb) [reply tests/AbstractTestCase.php:3]: added

File: tests/AnzuTestKernel.php (hunk 5 above)
Comment (sakulb) [reply tests/AnzuTestKernel.php:3]: added

**Pull request merged**